code/aepcomposeui/src/main/java/com/adobe/marketing/mobile/aepcomposeui/interactions/UIEevent.kt

```diff
@@ -0,0 +1,49 @@
+package com.adobe.marketing.mobile.aepcomposeui.interactions
+
+import com.adobe.marketing.mobile.aepuitemplates.AEPUITemplate
+import com.adobe.marketing.mobile.aepcomposeui.aepui.AEPUI
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.AEPUIState
+
+/**
+ * Sealed interface representing different types of UI events in AEP.
+ *
+ * @param T The type of UI template associated with the event.
+ * @param S The type of UI state associated with the event.
```

Comment:
`type of UI template/UI state` does not really describe how the template or state is related to the event produced here. Can we change it to something like this `AepUITemplate instance/ AepUiState instance backing the Composable on which the event has occurred`